### DISCUSSION TITLE
Multi sprite source

### DISCUSSION THREAD

**Discussion by IncidGeo** [maplibre/maplibre-gl-js#3962] [Q&A] [answered]
Hi everybody and thanks for this great library !
I try to work with multi sprite source in a json style, but is till get an error "Error: sprite: string expected, array found"

Here is an unique sprite that is working :
```
{
	"version": 8,
	"name": "custom",
	"glyphs":"http://localhost/static/carto/Fonts/{fontstack}/{range}.pbf",
	"sprite": "http://localhost/static/carto/sprite/Plan",
	
	"metadata": {
		"maputnik:renderer": "mbgljs"
	},
```
	
I've tried the multisprite source in conjunction with Martin :

```
{
	"version": 8,
	"name": "custom",
	"glyphs":"http://localhost/static/carto/Fonts/{fontstack}/{range}.pbf",
	"sprite": [
		{
			"id" : "default", 
			"url" : "http://localhost/static/carto/sprite/Plan"
		},
		{
			"id" : "test", 
			"url" : "http://localhost:3000/test"
		}
	]
	,
	"metadata": {
		"maputnik:renderer": "mbgljs"
	},
```
This code doesn't work either :
```

{
	"version": 8,
	"name": "custom",
	"glyphs":"http://localhost/static/carto/Fonts/{fontstack}/{range}.pbf",
	"sprite": [
		{
			"id" : "default", 
			"url" : "http://localhost/static/carto/sprite/Plan"
		}
	]
	,
	"metadata": {
		"maputnik:renderer": "mbgljs"
	},
```
I'm stuck, what am i doing wrong ? 
maplibre-gl-4.0.2

**Comment by HarelM** [comment]
Can you clarify on what didn't work? once the sprite is defined you need to reference it in a symbol layer with the relevant prefix.
Can you link to a jsbin/codepen/stackblitz reproduction?

**Comment by IncidGeo** [comment]
Yes there is a reference in the layer and it is working perfectly when there is a unique URL to the sprite like the first lines of the code on my first post :
```
"layout": {
				"visibility": "visible",
				"icon-image": "Barriere",
```
I've also tried with 
`"id" : "myid"`
and 
```
"layout": {
				"visibility": "visible",
				"icon-image": "myid:Barriere",
```

 instead of default but i don't think it's related.
I don't know what is not working only the error message ""Error: sprite: string expected, array found"' in the console when using multi-sprites source syntax.

**Comment by HarelM** [comment]
As requested before, please provide a jsbin/codepen/stackblitz, these partial examples are not helpful.

**Reply by IncidGeo** [reply]
I've never done that, but i'll try

**Reply by IncidGeo** [reply, marked as answer]
OK i've just upload (only change in code) the last version of the library 4.1.2 (was using 4.0.2) and everything now seems fine. 
I've tried to reproduce the error on stackblitz with the 4.0.2 library vs 4.1.2, but i' have not encountered the problem with a simple test code.
So no explanations but not a problem anymore.
https://stackblitz.com/edit/stackblitz-starters-eaimes?file=index.html
